Use `github_api` in sarif upload script

wzieba opened the pull request:
---

- [ ] I have considered if this change warrants release notes and have added them to the appropriate section in the `CHANGELOG.md` if necessary.

wzieba committed:
Use `github_api` in sarif upload script

wzieba commented on bin/upload_sarif_to_github:
```diff
@@ -62,10 +62,4 @@ fi
-  -H "X-GitHub-Api-Version: 2022-11-28" \
-  --data-binary "@$sarif_json_temp_file" \
-  https://api.github.com/repos/"$slug"/code-scanning/sarifs
+github_api "https://api.github.com/repos/$slug/code-scanning/sarifs" --data-binary "@$sarif_json_temp_file"
```
It still requires `$slug`, but not rest of the fields.

I intentionally removed `X-GitHub-Api-Version: 2022-11-28` - I don't think it's actually necessary

WDYT @AliSoftware

AliSoftware replied:
According to its code-comment doc and implementation, I think the first parameter of the `github_api` call is expected to only be the path (endpoint), so I'd expect this to be instead:
```suggestion
github_api "repos/$slug/code-scanning/sarifs" --data-binary "@$sarif_json_temp_file"
```

AliSoftware replied:
I just noticed that your code used `Authorization: Bearer $GITHUB_TOKEN` while the `github_api` helper uses `Authorization: token $GITHUB_TOKEN`. I wonder if there's a difference, or if both of those are equivalent and interchangeable… 🤔

AliSoftware replied:
[Answer to the above comment of mine](https://docs.github.com/en/rest/authentication/authenticating-to-the-rest-api?apiVersion=2022-11-28):

> Note: In most cases, you can use `Authorization: Bearer` or `Authorization: token` to pass a token. However, if you are passing a JSON web token (JWT), you must use `Authorization: Bearer`.

Maybe we should update `github_api` to use `Authorization: Bearer` while at it then, so that it is more generic and can cover all cases? 🤷

wzieba replied:
> I think the first parameter of the github_api call is expected to only be the path (endpoint), so I'd expect this to be instead:

Yeah, of course, sorry. I moved to quickly between contexts 😅 Fixed.

> Maybe we should update github_api to use Authorization: Bearer while at it then, so that it is more generic and can cover all cases? 🤷

Yeah, sounds like a good direction! 85fa9f2

wzieba committed:
Use the correct path for `github_api` execution

wzieba committed:
Use `Bearer` for auth header in `github_api`

https://github.com/Automattic/a8c-ci-toolkit-buildkite-plugin/pull/128#discussion_r1766804284

AliSoftware reviewed (approved):
You should also update the `CHANGELOG.md`; probably no need to add new entries/bullet points, just add the `[#128]` PR reference next to the 2 existing bugfix entries (in addition to them already referencing `[#124]`/`[#127]` respectively) should be enough?

Apart from that, LGTM 👍

wzieba committed:
Update changelog
